androidTest/java/com/owncloud/android/authentication/AuthenticatorActivityTest.java

```diff
@@ -88,8 +100,19 @@ public void check_login() {
                 .perform(typeText(BuildConfig.TEST_PASSWORD), closeSoftKeyboard());
         onView(withId(R.id.buttonOK)).perform(click());
 
-        // Check that login button is now enabled
-        onView(withId(R.id.buttonOK)).check(matches(isEnabled()));
+        // Check that the Activity ends after clicking
+        try {
+
+            Thread.sleep(WAIT_LOGIN);
+            Field f = Activity.class.getDeclaredField("mResultCode");
+            f.setAccessible(true);
+            mResultCode = f.getInt(mActivityRule.getActivity());
+
+        } catch (Exception e){
+            e.printStackTrace();
```

Comment:
Instead of catching any exception, I think would be better adding ```throws InterruptedException, NoSuchFieldException, IllegalArgumentException```to the test signature. That would make the test finish with a more specific error.  I think the stacktrace is not visible if printed directly, because the test apps filters logs.